src/os_unix.h

```diff
@@ -210,6 +210,9 @@ typedef struct dsc$descriptor   DESC;
 #ifndef SYS_VIMRC_FILE
 # define SYS_VIMRC_FILE "$VIM/vimrc"
 #endif
+#ifndef XDG_VIMRC_FILE
+# define XDG_VIMRC_FILE "~/.config/vim/vimrc"
```

Comment:
This doesn't even use the `$XDG_CONFIG_HOME` environment variable.

Reply:
@zeertzjq yes, I couldn't figure out how to do that yet, first time ever making a change in the vim codebase.

Reply:
@zeertzjq what would you advice? I choose to hard code it for now.

Reply:
@zeertzjq what would you advice?

Reply:
@zeertzjq what would you suggest?

Reply:
First, as mentioned in the issue, this should come after `USR_VIMRC_FILE2` etc..

And you can use `mch_getenv()` to check it `XDG_CONFIG_HOME` environment is set. If not, using `~/.config/vim/vimrc`, otherwise use `$XDG_CONFIG_HOME/vim/vimrc`.

Reply:
@zeertzjq Thanks for your feedback, I think the `ifdef` code around this section can be difficult to reason with (Neovim already refactored it I think?).

Anyway, unfortunately I'm a bit short on time at the moment and so I will leave it to someone more experienced to deal with.

Thanks.